velox/exec/MergeJoin.cpp

```diff
@@ -72,6 +80,12 @@ void MergeJoin::initialize() {
     }
   }
 
+  if (joinNode_->isLeftSemiFilterJoin()) {
```

Comment:
should we have the same check for right semi joins?

Reply:
@pedroerp Yes. Added [here](https://github.com/facebookincubator/velox/pull/9877/files#diff-8279da91a596a4b97a73bc4d9aa635da1baaf254dcdb2f7f8049bd21e7a9551eR69).